modules/host/host.go

```diff
@@ -343,18 +126,41 @@ type Host struct {
 	resourceLock sync.RWMutex
 }
 
+// checkUnlockHash will check that the host has an unlock hash. If the host
+// does not have an unlock hash, an attempt will be made to get an unlock hash
+// from the wallet. That may fail due to the wallet being locked, in which case
+// an error is returned.
+func (h *Host) checkUnlockHash() error {
```

Comment:
it's used to check whether the host has an unlock hash, and if the host doesn't have one then it gives the host an unlock hash.

I think `checkUnlockHash` as a name makes sense, though I understand the concerns about the state-change.